torch/fx/subgraph_rewriter.py

```diff
@@ -55,19 +72,28 @@ def attributes_are_equal(pn : Node, gn : Node) -> bool:
 
         # Traverse the use-def relationships to ensure that `pn` is a true
         # match for `gn`
-        if (pn.op != "output"
-                and len(pn.all_input_nodes) != len(gn.all_input_nodes)):
-            return False
-        match_found = all(self._match_nodes(pn_, gn_) for pn_, gn_
-                          in zip(pn.all_input_nodes, gn.all_input_nodes))
+        match_found = False
+        if pn.op == "output":
+            # This covers the case in which the original graph Node that
+            # matched to the pattern subgraph's output has more than
+            # one input Node
```

Comment:
(also discussing offline)

If I understand this correctly, it's basically:

1. We have an assumption that the pattern graph has exactly one output value
2. In the case that we have `pn` is the pattern output node and `gn` is some anchor, we need to check for the possibility of a pattern match on any of the nodes that `gn` consumes.


Are these assumptions correct? In that case I think we should be doing something like:

```
if pn.op == 'output':
  assert len(pn.all_input_nodes) == 1
  pattern_output_val = pn.all_input_nodes[0]
  match_found = any(self._match_nodes(pattern_output_val, gn_) for gn_ in gn.all_input_nodes)
<...>
```